src/librustc_middle/mir/mod.rs

```diff
@@ -81,11 +81,19 @@ impl<'tcx> HasLocalDecls<'tcx> for Body<'tcx> {
 pub enum MirPhase {
     Build = 0,
     Const = 1,
-    Validated = 2,
-    /// After this phase, the only `AggregateKind`s allowed are `Array` and `Generator`
-    DropElab = 3,
+    /// This phase checks the MIR for promotable elements and takes them out of the main MIR body
+    /// by creating a new MIR body per promoted element. After this phase (and thus the termination
+    /// of the `mir_promoted` query), these promoted elements are available in the `promoted_mir`
+    /// query.
+    ConstPromotion = 2,
+    /// After this phase
+    /// * the only `AggregateKind`s allowed are `Array` and `Generator`,
+    /// * `DropAndReplace` is gone for good
+    /// * `Drop` now uses explicit drop flags visible in the MIR and reaching a `Drop` terminator
+    ///   means that `Drop::drop` (or at least the generated drop glue) will be invoked.
```

Comment:
```suggestion
    ///   means that the auto-generated drop glue will be invoked.
```
This never invokes `Drop::drop` directly.